tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TypeSpec/CustomizedCodeUpdateTool.cs

```diff
@@ -190,25 +204,110 @@ private async Task<CustomizedCodeUpdateResponse> UpdateCoreAsync(string commitSh
         List<string> guidance,
         CancellationToken ct)
     {
-        var validationResult = await languageService.ValidateAsync(packagePath, ct);
-
-        if (validationResult.Success)
+        logger.LogInformation("ValidateAndUpdateGuidanceAsync called with packagePath: {PackagePath}", packagePath ?? "(null)");
+        
+        // First try to build the code to ensure it compiles after patch application
+        // This validates that the regenerated code with applied patches builds successfully
+        var buildResult = await BuildSdkAsync(packagePath, languageService, ct);
+        
+        if (buildResult.ExitCode == 0)
         {
-            logger.LogInformation("Validation passed");
-            guidance.Insert(0, "Code regenerated and validated successfully after applying patches.");
-            guidance.Insert(1, "");
-            return (true, false);
+            logger.LogInformation("Build completed successfully");
+            
+            // If build succeeds, also run language-specific validation if available
+            var validationResult = await languageService.ValidateAsync(packagePath, ct);
+            
+            if (validationResult.Success)
+            {
+                logger.LogInformation("Build and validation passed");
+                guidance.Insert(0, "Code regenerated, built, and validated successfully after applying patches.");
+                guidance.Insert(1, "");
+                return (true, false);
+            }
+            else
+            {
+                logger.LogWarning("Build succeeded but validation failed: {Errors}", string.Join(", ", validationResult.Errors));
+                guidance.Insert(0, "Code regenerated and built successfully after applying patches, but validation failed.");
+                guidance.Insert(1, $"Validation warnings: {string.Join(", ", validationResult.Errors)}");
+                guidance.Insert(2, "");
+                return (true, true); // Build succeeded but validation had issues
+            }
         }
         else
         {
-            logger.LogWarning("Validation failed: {Errors}", string.Join(", ", validationResult.Errors));
-            guidance.Insert(0, "Code regenerated but validation failed after applying patches.");
-            guidance.Insert(1, $"Validation errors: {string.Join(", ", validationResult.Errors)}");
+            logger.LogError("Build failed: {Error}", buildResult.ResponseError);
+            guidance.Insert(0, "Code regenerated but build failed after applying patches.");
+            guidance.Insert(1, $"Build errors: {buildResult.ResponseError}");
             guidance.Insert(2, "");
             return (false, true);
         }
     }
 
+    /// <summary>
+    /// Builds the SDK project to validate compilation.
+    /// </summary>
+    private async Task<PackageOperationResponse> BuildSdkAsync(string packagePath, LanguageService languageService, CancellationToken ct)
```

Comment:
We might move this method into the language service later to eliminate duplication.